dockstore-webservice/src/test/java/io/dockstore/webservice/languages/CWLHandlerTest.java

```diff
@@ -448,27 +428,23 @@ public void testLanguageVersionExtraction() throws IOException {
         CWLHandler cwlHandler = new CWLHandler();
 
         final String resourceRoot = "multi-version-cwl";
-        final SourceFile manyFile = mockMethodSetTypeVersion(mockSourceFile(resourceRoot, "/many-version.cwl"), v -> manyVersion = v);
-        final SourceFile oneFile = mockMethodSetTypeVersion(mockSourceFile(resourceRoot, "/one-version.cwl"), v -> oneVersion = v);
-        final SourceFile noFile = mockMethodSetTypeVersion(mockSourceFile(resourceRoot, "/no-version.cwl"), v -> noVersion = v);
+        final SourceFile manyFile = createSourceFile(resourceRoot, "/many-version.cwl");
+        final SourceFile oneFile = createSourceFile(resourceRoot, "/one-version.cwl");
+        final SourceFile noFile = createSourceFile(resourceRoot, "/no-version.cwl");
 
         // Test multiple files containing many versions.
-        manyVersion = "bogus";
-        oneVersion = "bogus";
-        noVersion = "bogus";
-        versionVersions = List.of("bogus");
-        final Version version = mockMethodSetDescriptorTypeVersions(Mockito.mock(Version.class), v -> versionVersions = v);
+        final Version version = new WorkflowVersion();
         cwlHandler.parseWorkflowContent(manyFile.getPath(), manyFile.getContent(), Set.of(manyFile, oneFile, noFile), version);
-        assertEquals("v1.2", manyVersion);
-        assertEquals("v1.1", oneVersion);
-        assertEquals(null, noVersion);
-        assertEquals(List.of("v1.2", "v1.1", "v1.0"), versionVersions);
+        assertEquals(List.of("v1.2", "v1.1", "v1.0"), version.getVersionMetadata().getDescriptorTypeVersions());
+        assertEquals("v1.2", manyFile.getMetadata().getTypeVersion());
+        assertEquals("v1.1", oneFile.getMetadata().getTypeVersion());
+        assertEquals(null, noFile.getMetadata().getTypeVersion());
```

Comment:
To be sure that the values we're reading were set by `parseWorkflowContent`, we might set the language versions of the `SourceFiles` to a bogus value beforehand.  This probably only really matters for `noFile`, so that we can confirm that the `null` value was set by `parseWorkflowContent`. rather than simply passing through unchanged.